notte/llms/prompt.py

```diff
@@ -21,21 +21,27 @@ def get(self, prompt_id: str) -> list[Message]:
             with open(prompt_file, "r") as file:
                 content: str = file.read()
                 role: str = prompt_file.name.split(".")[0]
-                messages.append(Message(role=role, content=content))
+                if role not in ["assistant", "user", "system", "tool", "function"]:
+                    raise ValueError(f"Invalid role: {role}")
+                messages.append(Message(role=role, content=content))  # type: ignore
         return messages
 
     def materialize(self, prompt_id: str, variables: dict[str, Any] | None = None) -> list[dict[str, str | None]]:
         # TODO. You cant pass variables that are not in the prompt template
         # But you can fewer variables than in the prompt template
         messages = self.get(prompt_id)
         if variables is None:
-            return messages
+            return self.dictify(messages)
 
         try:
             materialized_messages: list[Message] = []
             for message in messages:
+                if message.content is None:
```

Comment:
Message content cannot be None here

Reply:
it can cf method in litelmm
<img width="517" alt="Screenshot 2024-12-11 at 10 35 16" src="https://github.com/user-attachments/assets/d93d3a18-9d7b-4a82-8682-3d438087d4eb" />